lib/libvmod_cookie/vmod.vcc

```diff
@@ -0,0 +1,248 @@
+$Module cookie 3 "Varnish Cookie Module"
+DESCRIPTION
+===========
+
+Handle HTTP cookies easier in Varnish VCL.
+
+Parses a cookie header into an internal data store, where per-cookie
+get/set/delete functions are available. A filter_except() method removes all
+but a set comma-separated list of cookies. A filter() method removes a comma-
+separated list of cookies.
+
+Regular expressions can be used for either selecting cookies, deleting matching
+cookies and deleting non-matching cookie names.
+
+A convenience function for formatting the Set-Cookie Expires date field
+is also included. If there are multiple Set-Cookie headers vmod-header
+should be used.
```

Comment:
I don't think an in-tree VMOD should reference an out-of-tree VMOD to deal with its shortcomings.

As much as I liked vmod_cookie to the point where I took inspiration from it for vmod_querystring, I ended up rewriting it from scratch (designing a new API taking advantage of new features from Varnish 4.1) to offer something both easier and more efficient to use.

Of course "easier" is subjective, but it resulted in something easier to implement, document and maintain (I'm looking at you, `filter_except`, and you know why).

I'm not suggesting we copy vmod_querysting to implement vmod_cookie but I'd rather start from a new VCC file. Once in tree, we can then rename vmod_cookie to something else like vmod_legacycookie in the varnish-modules project and end-users can keep using it with minimal changes in their existing VCL:

```diff
-import cookie;
+import legacycookie as cookie;
```

Also, a mistake happened at some point: `vmod.c` is a copy of `vmod.vcc`.

This module comes with a ton of tests and the current namespace we have makes it increasingly harder to figure where to find existing or put new VMOD coverage. With this many new test cases I'd rather start a new namespace for test cases dedicated to VMOD functionality coverage. We can introduce that change with vmod_cookie and then move other in-tree tests for the rest of our VMODs.

Of course it doesn't mean we can't use VMODs in test cases from the current namespace.

An exception I would make is vmod_vtc and vmod_debug that are only meant to support the test suite.

As we are increasing the number of in-tree modules I'd also like to stop using `vmod.vcc` in favor of `vmod_*.vcc`.